### ISSUE TITLE
Is this code ever tested?

### ISSUE THREAD

**Issue by sky8273** [thomerow/usb-hid-library#1]
Searching for C# USB hid code and this one looks promising.
But this is no guideline about  it .
And i found this line doesn't work. 

```
        // Store the report data following the report ID.
        Array.ConstrainedCopy(data, 0, outputReportBuffer, 1, outputReportBuffer.Length);
```

It looks like the variable "HIDReportLength" is not defined clearly. data.Length equals outputReportBuffer.Length.

Is this lib code ever tested? 
I really appreciate if there is a fully functioning version of this lib.

**Comment by thomerow** [comment]
You are right, that shouldn't work. I guess I didn't get any errors while using the library becaus the project I wrote it for only needed to read from the USB device and never sent any data to it.

`HIDReportLength` is defined further up as `private const int HIDReportLength = 65;`.

The only thing I did with the library was connecting a big red Buzzer knob to the USB port via a microcontroller. Essentially it only set one bit of the whole 65 byte HID data packet to one or zero to indicate its state (pressed / not pressed). :-D 

For that purpose it did its job perfectly well. Otherwise it is pretty much untested.
